### PR TITLE
feat(client): support last middleware for client

### PR DESCRIPTION
**Pull request by welkeyever**
#### What type of PR is this?
feat
#### Check the PR title.
- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
Client 支持设置一个last中间件，主要用于对于最终发起请求前的数据最终处理操作

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: 上下文中间件会将一些 ctx 转换为 http 的 header 信息，但如果 ctx 中间件位置过于靠前（往往没法选择），将导致用户后来新增的中间件中针对 ctx 元数据的修改等操作无法生效到最终的header中来，因此增加一个配置最终中间件的方式，让这类操作有机会在用户的中间件执行之后最终收尾。
zh(optional): The context middleware will convert some ctx to http header information, but if the ctx middleware is located too high (often unselectable), it will cause the modification of ctx metadata in the middleware added later by the user to fail to take effect Come to the final header, so add a way to configure the final middleware, so that this type of operation has the opportunity to be finalized after the user's middleware is executed.

#### Which issue(s) this PR fixes:
None

### CODE REVIEW

**Review by Duslia — DISMISSED** [review]

File: pkg/app/client/client.go [hunk 1]
```diff
@@ -616,6 +620,11 @@ func (c *Client) Use(mws ...Middleware) {
 	c.mws = chain(middlewares...)
 }
 
+// UseAsLast is used to add middleware to the end of the middleware chain.
+func (c *Client) UseAsLast(mw Middleware) {
+	c.lastMiddleware = mw
```
Comment (FGYFFFF) [inline pkg/app/client/client.go:625]: 这个每次调用都会被覆盖，那么用户调用多次后怎么办？一些逻辑就会被覆盖掉？

**Comment by welkeyever** [comment]
Possibly add the .GetLastMiddleware() which will let users have the chance to get the last Middleware and chain it explicitly.

File: pkg/app/client/client.go (hunk 1 above)
Comment (welkeyever) [reply pkg/app/client/client.go:625]: 好问题，可能的策略是UseAsLast默认判断是否存在已配置的last middleware（这个同样也是一个额外的上下文：业务显示感知已有一个需要在末尾执行的中间件），配合下面提到的 GetXXX接口能够将其取出，按照需求在其前后按需链进去。

比较尴尬的一个点是，框架没办法帮助用户判断（如果存在多个需要在last位置执行的中间件最终的优先级），因此提供上述机制让用户自行选择：这部分如果有更好的方案也可以直接抛出来

**Comment by welkeyever** [comment]
> Possibly add the .GetLastMiddleware() which will let users have the chance to get the last Middleware and chain it explicitly.

Add `TakeOutLastMiddleware` to let user decide the final order of the middleware.

File: pkg/app/client/client.go [hunk 2]
```diff
@@ -616,6 +623,28 @@ func (c *Client) Use(mws ...Middleware) {
 	c.mws = chain(middlewares...)
 }
 
+// UseAsLast is used to add middleware to the end of the middleware chain.
+//
+// Will return an error if last middleware has been set before, to ensure all middleware has the change to work,
+// please use `TakeOutLastMiddleware` to take out the already set middleware.
```
Comment (FGYFFFF) [inline pkg/app/client/client.go:629]: please -> Please

File: pkg/app/client/client.go [hunk 3]
```diff
@@ -616,6 +623,28 @@ func (c *Client) Use(mws ...Middleware) {
...
+	if c.lastMiddleware != nil {
+		return errorLastMiddlewareExist
+	}
+	c.lastMiddleware = mw
+	return nil
+}
+
+// TakeOutLastMiddleware will return the set middleware and remove it from client.
+//
+// Remember to Set it back after chain it with other middleware.
```
Comment (FGYFFFF) [inline pkg/app/client/client.go:641]: Set -> set

Comment (welkeyever) [reply pkg/app/client/client.go]: done

**Review by FGYFFFF — APPROVED** [review]

**Review by wzekin — APPROVED** [review]

**Review by Duslia — APPROVED** [review]

**Pull request merged**